### PR TITLE
Rename panoptic seg related and Realse instance seg related for Mask2Former

### PR DESCRIPTION
**Pull request by chhluo**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

1. Rename configs for panoptic segmentation.
2. Release configs, logs and checkpoints for instance segmentation. For instance segmentation, We have trained the models for many times with the algorithm we implemented (see [PR 7571](https://github.com/open-mmlab/mmdetection/pull/7571)), and the performance of the trained models is relatively stable (+- 0.2), and there is a certain gap between the performance of the models we trained and the results mentioned in the [paper](http://arxiv.org/abs/2112.01527). However, the performance of the model trained with the official code are unstable, and the performance may be lower than the target results (see [issue](https://github.com/facebookresearch/Mask2Former/issues/46)).
3. fix bug in gather_models.py

## Modification

1. Rename configs for panoptic segmentation: `mask2former_xxx_coco.py` -> `mask2former_xxx_coco-panoptic.py`. `mask2former_xxx_coco.py` is for instance segmentation, while `mask2former_xxx_coco-panoptic.py` for panoptic segmentation.

## BC-breaking (Optional)

Rename configs for panoptic segmentation: `mask2former_xxx_coco.py` -> `mask2former_xxx_coco-panoptic.py`. `mask2former_xxx_coco.py` is for instance segmentation, while `mask2former_xxx_coco-panoptic.py` for panoptic segmentation.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
3. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
5. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
6. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by ZwwWayne — APPROVED** [review]

File: configs/mask2former/README.md [hunk 1]
```diff
@@ -38,15 +38,28 @@ mmdetection
...
+
+Note: We have trained the above models(instance segmentation) many times with the algorithm we implemented (see [PR 7571](https://github.com/open-mmlab/mmdetection/pull/7571)), and the performance of the trained models is relatively stable (+- 0.2), and there is a certain gap between the performance of the models we trained and the results mentioned in the [paper](http://arxiv.org/abs/2112.01527). However, the performance of the model trained with the official code is unstable, and the performance may be lower than the target results (see [issue](https://github.com/facebookresearch/Mask2Former/issues/46)).
```
Comment (ZwwWayne) [inline configs/mask2former/README.md:62]: **Note:** We have trained the instance segmentation models many times (see more details in [PR 7571](https://github.com/open-mmlab/mmdetection/pull/7571)). The results of the trained models are relatively stable (+- 0.2), and have a certain gap in comparison with the results in the [paper](http://arxiv.org/abs/2112.01527). However, the performance of the model trained with the official code is unstable and may also be slightly lower than the reported results as mentioned in the [issue](https://github.com/facebookresearch/Mask2Former/issues/46).

File: configs/mask2former/README.md [hunk 2]
```diff
@@ -59,7 +59,7 @@ mmdetection
...
 
-Note: We have trained the above models(instance segmentation) many times with the algorithm we implemented (see [PR 7571](https://github.com/open-mmlab/mmdetection/pull/7571)), and the performance of the trained models is relatively stable (+- 0.2), and there is a certain gap between the performance of the models we trained and the results mentioned in the [paper](http://arxiv.org/abs/2112.01527). However, the performance of the model trained with the official code is unstable, and the performance may be lower than the target results (see [issue](https://github.com/facebookresearch/Mask2Former/issues/46)).
+Note: We have trained the instance segmentation models many times (see more details in [PR 7571](https://github.com/open-mmlab/mmdetection/pull/7571)). The results of the trained models are relatively stable (+- 0.2), and have a certain gap in comparison with the results in the [paper](http://arxiv.org/abs/2112.01527). However, the performance of the model trained with the official code is unstable and may also be slightly lower than the reported results as mentioned in the [issue](https://github.com/facebookresearch/Mask2Former/issues/46).
```
Comment (ZwwWayne) [inline configs/mask2former/README.md:62]: and have a certain gap -> and have a certain gap (about 0.x AP)

File: configs/mask2former/README.md [hunk 3]
```diff
@@ -38,15 +38,28 @@ mmdetection
...
+| Swin-S   | -       | ImageNet-1K | 50e     | 18.8     | -              | 49.3    | 46.1     | [config](https://github.com/open-mmlab/mmdetection/blob/master/configs/mask2former/mask2former_swin-s-p4-w7-224_lsj_8x2_50e_coco.py) | [model](https://download.openmmlab.com/mmdetection/v2.0/mask2former/mask2former_swin-s-p4-w7-224_lsj_8x2_50e_coco/mask2former_swin-s-p4-w7-224_lsj_8x2_50e_coco_20220504_001756-743b7d99.pth) \| [log](https://download.openmmlab.com/mmdetection/v2.0/mask2former/mask2former_swin-s-p4-w7-224_lsj_8x2_50e_coco/mask2former_swin-s-p4-w7-224_lsj_8x2_50e_coco_20220504_001756.log.json) |
+
+Note: We have trained the instance segmentation models many times (see more details in [PR 7571](https://github.com/open-mmlab/mmdetection/pull/7571)). The results of the trained models are relatively stable (+- 0.2), and and have a certain gap (about 0.2 AP) in comparison with the results in the [paper](http://arxiv.org/abs/2112.01527). However, the performance of the model trained with the official code is unstable and may also be slightly lower than the reported results as mentioned in the [issue](https://github.com/facebookresearch/Mask2Former/issues/46).
```
Comment (ZwwWayne) [inline configs/mask2former/README.md:62]: and and have -> and have

**Comment by Levine66** [comment]
Hey, thanks for all of your work for Mask2Former. Recently, i am learning the code of [MPFomer,](https://github.com/IDEA-Research/MP-Former) which is modified from Mask2Former, and the only differences between MPFormer and Mask2Fomer are in mask2former_transformer_decoder.py and criterion.py which construct the  MP training queries and loss function. I would like to convert the code from Detectron2 to MMDetection, but i am confused that I can't find the part(mask2former_transformer_decoder.py) they modified in MMDetection, it makes me hard to rewrite the code in MMDetection. So i would like to ask you guys to give me some advices to implement thisk work your time permits.

**Comment by chhluo** [comment]
@Levine66, Code related to transformer decoder is in https://github.com/open-mmlab/mmdetection/blob/2.x/mmdet/models/utils/transformer.py.

**Pull request merged**